<launch>

    <!-- arguments -->
    <arg name = "filter"             value = "true" />
    <arg name = "cov"                value = "true" />
 
    <arg name = "imu"                value = "/d435i/imu/data" />
    <arg name = "lidar_slam_odom"    value = "/ekf_loam/laser_odom_to_initOut" />
    <arg name = "tracking_camera"    value = "/t265/odom/sample" />
    <arg name = "visual_slam_odom"   value = "/rtabmap/odom" />
    <arg name = "wheel_odom"         value = "/odom" />
    <!-- <arg name = "ptCloud"            value = "/os1_cloud_node/points" /> -->
    <arg name = "stereo_cam_left"    value = "/t265/fisheye2/image_raw" />
    <arg name = "stereo_cam_right"   value = "/t265/fisheye1/image_raw" />
    <arg name = "color_image"        value = "/d435i/color/image_raw" />
    <arg name = "lidar_odom_out"     value = "/ekf_loam/laser_odom_to_init_cov" />
    <arg name = "visual_odom_out"    value = "/visual/odom/sample_cov" />
    <arg name = "wheel_odom_out"     value = "/odom_cov" />
    <arg name = "rtabmap_service"    value = "/rtabmap/reset_odom_to_pose" />
    
    <rosparam command="load" file="$(find adaptive_odom_filter)/config/adaptive_filter_parameters.yaml" />
    
    <!-- State Machine -->
    <node if="$(arg cov)"  pkg="adaptive_odom_filter"  type="EKFAdaptiveFilter"  name="EKFAdaptiveFilter"  output="screen">
        <remap from="/imu/data"                        to="$(arg imu)"/>
        <remap from="/ekf_loam/laser_odom_to_initOut"  to="$(arg lidar_slam_odom)"/>
        <remap from="/t265/odom/sample"                to="$(arg tracking_camera)"/>
        <remap from="/rtabmap/odom"                    to="$(arg visual_slam_odom)"/>
        <remap from="/odom"                            to="$(arg wheel_odom)"/>
        <!-- <remap from="/os1_cloud_node/points"           to="$(arg ptCloud)"/> -->
        <remap from="/t265/fisheye2/image_raw"         to="$(arg stereo_cam_left)"/>
        <remap from="/t265/fisheye1/image_raw"         to="$(arg stereo_cam_right)"/>
        <remap from="/d435i/color/image_raw"           to="$(arg color_image)"/>
        <remap from="/ekf_loam/laser_odom_to_init_cov" to="$(arg lidar_odom_out)"/>
        <remap from="/visual/odom/sample_cov"          to="$(arg visual_odom_out)"/>
        <remap from="/odom_cov"                        to="$(arg wheel_odom_out)"/>
        <remap from="/rtabmap/reset_odom_to_pose"      to="$(arg rtabmap_service)"/>
    </node> 
    

</launch>
